src/cpp/src/llm_pipeline.cpp

```diff
@@ -206,21 +209,53 @@ class LLMPipeline::LLMPipelineImpl {
                         "(input_ids, attention_mask, position_ids, beam_idx) "
                         "but you have '" + std::to_string(num_inputs) + "' inputs");
 
+        
+        size_t kv_cache_len = 0;
+        ov::Tensor atten_mask;
+        if (is_chat_conversation && !m_is_cache_empty) {
+            OPENVINO_ASSERT(batch_size == 1, "continuation of generation is possible only for batch 1");
+            // If history is saved in KV cache concatenate new attention_mask with the already existing.
+            // Between subsequent runs attention_mask should not be modified.
+            auto atten_mask_history = m_model_runner.get_tensor("attention_mask");
+            auto prompt_len = attention_mask.get_shape()[1];
+            kv_cache_len = atten_mask_history.get_shape()[1];
+
+            ov::Tensor new_atten_mask =  ov::Tensor{ov::element::i64, {batch_size, kv_cache_len + prompt_len}};
+            std::copy(atten_mask_history.data<int64_t>(), atten_mask_history.data<int64_t>() + kv_cache_len,
+                    new_atten_mask.data<int64_t>());
+            std::copy(attention_mask.data<int64_t>(), attention_mask.data<int64_t>() + prompt_len,
+                    new_atten_mask.data<int64_t>() + kv_cache_len);
+            atten_mask = new_atten_mask;
+        } else {
+            atten_mask = attention_mask;
+        }
```

Comment:
attention mask and position ids are specific for stateful and static model runners, while they are not required for PA-based.
I would move it to model runners / pipelines to avoid complexity in generic LLM pipeline impl.

But maybe it should be done during migration to CB rails
CC @Wovchena

Reply:
There's no other way for now